docs/application/dotnet/guides/applications/component-based-app.md

```diff
@@ -0,0 +1,607 @@
+            <icon>org.tizen.sample.png</icon>
+            <label>base-widget</label>
+            <label xml:lang="en-us">base-widget</label>
+            <label xml:lang="ko-kr">base-widget[KOR]</label>
+            <support-size>2x2</support-size>
+        </widget-component>
+    </component-based-application>
+    <privileges>
+        <privilege>http://tizen.org/privilege/appmanager.launch</privilege>
+    </privileges>
+</manifest>
+```
+
+Following are the main attributes:
+
+- `id`
+
+  Every component element has an `id` attribute. The attribute ID must be globally unique in a device. If not, the application installation fails. This attribute is also used by other applications to confirm which component must be created.
+
+- `main`
+
+  Component based applications can have multiple components and there is one main component. The main component is the component that is created when the component based application receives a launch request without a specific component ID. To set the main component, set the `main` attribute to `true`.
+
+- `icon-display`
+
+  This attribute is only for the frame component element. If this value is set to `true`, the home application will display an icon on the app tray for the frame components.
+
+- `launch_mode`
+
+  This attribute is only for the frame component element. Application launch mode values are:
+  - Single: Launched as the main application.
+  - Group: Launched as a sub-application.
+  - Caller: Defines the launch mode with `AppControl.SetLaunchMode()`.
+
+- `taskmanage`
+
+  This attribute is only for the frame component element. This attribute indicates whether the application appears in the task manager or not. The available values are `true` and `false`.
+
+- `setup-appid`
+
+  This attribute is only for the widget component element.
+  This attribute is app id that used when launch widget component element.
+
+- `max-instance`
+  This attribute is only for the widget component element.
+  This attribute limits the number of widget instances concurrently executable for a Web application. When omitted or its value is 0, unlimited number of widget instances are supported. The expected value is integer.
+
+## Prerequisites
+
+To use the functions and data types of the component based application API (in [mobile](../../api/mobile/latest/group__COMPONENT__BASED__APPLICATION__MODULE.html) and [wearable](../../api/wearable/latest/group__COMPONENT__BASED__APPLICATION__MODULE.html) applications), include the `Tizen.Applications.ComponentBased.Common` module in your application:
+
+```
+using Tizen.Applications.componentbased.common
+using Tizen.Applications.componentbased.default  // if you want to use ELFComponentBasedApplication
+```
+
+<a name="application_monitoring"></a>
+## Start Component Based Application
+
+To start a component based application, you must register the component based application callbacks and start a main event loop as shown in the following code:
+
+```
```

Comment:
For code blocks you need to mention the programming language, for reference see
https://github.com/Samsung/tizen-docs/edit/master/docs/application/dotnet/guides/machine-learning/pipeline.md

Reply:
I added it. Thanks